src/presentation/brokers/consumers/consumer.ts

```diff
@@ -0,0 +1,16 @@
+export default class BrokerConsumer {
+  private broker: any;
+
+  constructor(broker: any) {
+    this.broker = broker;
+  }
+
+  public async consume() {
+    const connection = await this.broker.connect();
+    const channel = await this.broker.connectToChannel(connection);
```

Comment:
I dont't think the `connectToChannel` method is needed at all.
Why not just do
```suggestion
    const channel = await connection.createChannel();
```